File: leetcode/april2024/08_number_of_students_unable_to_eat.cpp
class Solution {
    public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int preferCircle = 0, preferSquare = 0;
        for(int s : students) {
            if(s == 0) preferCircle++;
            else preferSquare++;
        }
        for(int s : sandwiches) {
            if(s == 0 and preferCircle == 0) 
                return preferSquare;
            if(s == 1 and preferSquare == 0)
                return preferCircle;
            
            if(s == 0) preferCircle--;
            else preferSquare--;
        }
        return 0;
    }
};

File: leetcode/april2024/python/08_number_of_students_unable_to_eat.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        preferCircle = preferSquare = 0
        for s in students:
            if s == 1: preferSquare += 1
            else: preferCircle += 1
        for s in sandwiches:
            if s == 0 and preferCircle == 0:
                return preferSquare
            if s == 1 and preferSquare == 0:
                return preferCircle  
            if s == 1: preferSquare -= 1
            else: preferCircle -= 1
        return 0  